model/src/lib.rs

```diff
@@ -45,41 +45,49 @@ pub struct UserOrder {
     pub valid_to: u32,
     pub app_data: u32,
     #[serde(with = "u256_decimal")]
-    pub tip: U256,
+    pub fee_amount: U256,
     pub order_kind: OrderKind,
     pub partially_fillable: bool,
     pub signature: Signature,
 }
 
-impl UserOrder {
+impl OrderCreation {
     pub fn token_pair(&self) -> Option<TokenPair> {
         TokenPair::new(self.buy_token, self.sell_token)
     }
 }
 
-/// An order that is returned when querying the orderbook.
-///
-/// Contains extra fields thats are populated by the orderbook.
-#[derive(Eq, PartialEq, Clone, Copy, Debug, Deserialize, Serialize)]
+/// An order as provided to the orderbook by the frontend.
+#[derive(Eq, PartialEq, Clone, Debug, Deserialize, Serialize)]
 #[serde(rename_all = "camelCase")]
-pub struct Order {
-    pub creation_time: DateTime<Utc>,
+pub struct OrderMetaData {
+    pub creation_date: DateTime<Utc>,
     #[serde(with = "h160_hexadecimal")]
     pub owner: H160,
-    #[serde(flatten)]
-    pub user_provided: UserOrder,
+    pub uid: String,
```

Comment:
I also think this should internally store fixed size bytes instead of string. A wrapping struct like what Nic suggested is nice because we can implement methods and serialization on it.
If the serialization is complicated for this PR feel free to do it in another.

Reply:
Unfortunately, I needed to reimplement all traits, because arrays of size bigger than 32 suck in rust, according to my limited knowledge :) 
If you guys know better, please! help.

Reply:
I don't think so. ```rust
#[derive(Eq, PartialEq, Clone, Copy, Debug)]
pub struct Foo([u8; 56]);
```
compiles.

Reply:
You might need to update your Rust version. Recently they've been rolling out "const generics" so most of the std library traits are all implemented `for<const N: usize> [u; N]`.